File: python/array/lcs.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]):
        res = 0
        numSet = set(nums)
        for num in numSet:
            if num - 1 not in numSet:
                streak = 1
                while num + 1 in numSet:
                    num += 1
                    streak += 1
                res = max(res, streak)
        return res

File: go/arr/lcs.go
package main

func longestConsecutive(nums []int) int {
	set := make(map[int]bool)

	for _, num := range nums {
		set[num] = true
	}
	maxSeq := 0
	for _, num := range nums {
		if _, ok := set[num-1]; !ok {
			currSeq := 0
			for set[num] {
				currSeq++
				num++
			}
			maxSeq = max(maxSeq, currSeq)
		}
	}
	return maxSeq
}
